### ISSUE TITLE
RFC: Web.HTTP の URL 関係を Web.URI に移動してほしい

### ISSUE THREAD

**Issue by lambdalisue** [vim-jp/vital.vim#376]
具体的には https://github.com/vim-jp/vital.vim/blob/master/doc/vital-web-http.txt#L113-L120 の辺

Web.HTTP の書き直し（or Network.HTTP の作成）で利用したいので URL 関係の処理は別モジュールにしたいと思うのですがどうでしょうか？

**Comment by thinca** [comment]
`URI.decode()` と `URI.encode()` はすでにあります。 `application/x-www-form-urlencoded` については、`URI` の扱う範疇を超えているので、ちょっと考える必要があります。

**Comment by lambdalisue** [comment]
> URI の扱う範疇を超えているので、ちょっと考える必要があります。

問題はそこなのですが `Web.HTTP` はリクエストだけを扱うほうが楽なんですよね。テストとかの関係上

**Comment by thinca** [comment]
Web.HTTP は廃棄の方向で再実装でもいい気もしてます。(どちらにしても `URI` のやつと微妙に動作がかぶるので利用者にわかりづらいというのはある)

**Comment by lambdalisue** [comment]
> Web.HTTP は廃棄の方向で再実装でもいい気もしてます。

はい、基本そのつもりでしたが実装重いので `Web.HTTP` から使えるものをほかモジュールに移行しつつ `Network.HTTP` の作成に着手したいと思い RFC しています。`Network.HTTP` はリクエストだけを扱いたい（URI 関係のテストは別に投げたい）

**Comment by lambdalisue** [comment]
@tyru さんどう思います？ (`Web.URI` 書いてる関係上)

そろそろ `System.Process` が固まりそうなので、こちらも固めたい。
個人的には `encodeURIComponent` も `Web.URI` にあったほうが楽ですが、`URI`の範疇超えてるし `Network.HTTP` でやるべきというならばがんばります。

c.f. @thinca

**Comment by tyru** [comment]
急いで入れて Changes の行を増やすのもアレなので、とりあえず今はサポート無しで良いんじゃないですかね。
ありすえさんが application/x-www-form-urlencoded の実装が今すぐ欲しいなら別ですが。

**Comment by lambdalisue** [comment]
急いでという意味ではありませんでした。実装を行うというよりかは最終形がどちらになるのかを固めたかったという感じです

**Comment by tyru** [comment]
了解です。
自分としては、@thincaさんも言ってる通りどこにどのメソッドを入れるか決めあぐねている状態なら、今は入れなくていいんじゃないかな、程度でした。

> application/x-www-form-urlencoded については、URI の扱う範疇を超えているので、ちょっと考える必要があります。
